### DISCUSSION TITLE
Record rules forn data sources

### DISCUSSION THREAD

**Discussion by saguadob** [grafana/grafana#48242] [Data Sources]
Now it is only possible to create record rules with data that is stored in Prometheus or Loki instances. 

It would be helpful to be able to create recording rules from external data sources that can be stored in Prometheus/or Loki. As an example, one could use an Azure Monitor query log and record a numeric value as a metric in a Prometheus instance.

This could potentially enable engineers to aggregate data from different data sources into Prometheus and create better alert definitions using cortex, etc.

Currently, engineers must create their own components that push data to Prometheus to achieve this; using the already the rich grafana data sources ecosystem could simplify the creation of efficient monitoring systems.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
